pkg/kube/client/ssa/state_cache.go

```diff
@@ -0,0 +1,107 @@
+package ssa
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+
+	xpresource "github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	objectv1alpha2 "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha2"
+)
+
+// StateCacheManager lets you manage StateCache entries for XP managed
+// resources
+type StateCacheManager interface {
+	LoadOrNewForManaged(mg xpresource.Managed) StateCache
+	Remove(mg xpresource.Managed)
```

Comment:
Is there any connection of this to the openapi/extractor caching? Can we put them into different packages?

Reply:
Yes, we can. There is no direct relation, but they are related in the context of provider-kubernetes's SSA-enabled client

Reply:
ssa/cache/state
ssa/cache/extractor

Would make it easier to understand.

Reply:
done